src/cloudfront/index.ts

```diff
@@ -141,13 +140,106 @@ export const createCloudFrontDistribution = async (
   await cloudfront
     .waitFor('distributionDeployed', { Id: Distribution.Id })
     .promise()
+
   return Distribution
 }
 
+const createAndPublishNoDefaultRootObjectRedirectionFunction = async () => {
+  const cloudFrontRedirectionFunctionName =
+    NO_DEFAULT_ROOT_OBJECT_REDIRECTION_FUNCTION_NAME +
+    '_' +
+    NO_DEFAULT_ROOT_OBJECT_REDIRECTION_COLOR
+
+    let functionARN = await isCloudFrontFunctionExisting(
+      cloudFrontRedirectionFunctionName
+    )
+
+  if (!functionARN) {
+    const { createdFunctionETag, createdFunctionARN } =
+      await createNoDefaultRootObjectFunction(cloudFrontRedirectionFunctionName)
+
+    if (createdFunctionETag) {
+      await publishCloudFrontFunction(
+        cloudFrontRedirectionFunctionName,
+        createdFunctionETag
+      )
+    }
+
+    if (!createdFunctionARN) {
+      throw new Error(
+        `[CloudFront] Could not create lambda function to handle redirection when no default root object`
+      )
+    }
+
+    functionARN = createdFunctionARN
+  }
+  return functionARN
+}
+
+const isCloudFrontFunctionExisting = async (name: string) => {
+  let existingFunctionARN: string | undefined
+  await cloudfront
+    .listFunctions((err, data) => {
+      if (err) {
+        logger.error(`[CloudFront] Error listing lambda functions: ${err}`)
+      }
+      existingFunctionARN = data.FunctionList?.Items?.find(item => item.Name === name)?.FunctionMetadata.FunctionARN
+    })
+    .promise()
+  return existingFunctionARN
+}
+
+const createNoDefaultRootObjectFunction = async (functionName: string) => {
+  logger.info(
+    `[CloudFront] ✏️ Creating lambda function to handle redirection when no default root object...`
+  )
+  let createdFunctionETag: string | undefined
+  let createdFunctionARN: string | undefined
+  const data = await cloudfront
+    .createFunction(
+      {
+        Name: functionName,
+        FunctionCode: noDefaultRootObjectFunctions[NO_DEFAULT_ROOT_OBJECT_REDIRECTION_COLOR],
+        FunctionConfig: {
+          Runtime: 'cloudfront-js-2.0',
+          Comment:
+            'Redirects to branch specific index.html when no default root object is set',
+        },
+      },
+      (err) => {
+        if (err) {
+          logger.error(`[CloudFront] Error creating lambda function: ${err}`)
```

Comment:
```suggestion
          logger.error(`[CloudFront] Error creating function: ${err}`)
```

Reply:
Don't we want to throw in this case ?

Reply:
There is a weird behavior here, AWS sends an error stating that the function already exists even if it successfully created it. It may be a bug or a feature, but we don't want to throw for now.